hazelcast-sql/pom.xml

```
@@ -419,6 +419,21 @@
                     </dependency>
                 </dependencies>
             </plugin>
+            <plugin>
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
+                <configuration>
+                    <excludes>
+                        <!--
+                        Needed for https://github.com/hazelcast/hazelcast/issues/20945.
+                        Jacoco injects methods such as this:
+                           private static boolean[] org.apache.calcite.rel.metadata.BuiltInMetadata$RowCount$Handler.$jacocoInit()
+                        This then breaks an assertion in org.apache.calcite.rel.metadata.MetadataDef.
+                        -->
+                        <exclude>**$Handler</exclude>
```

You can make this more specific so it doesn't capture every `Handler` inner class, but only e.g. in `org.apache.calcite.rel.metadata` package:
```suggestion
                        <exclude>org/apache/calcite/rel/metadata**$Handler</exclude>
```

---

Missed this comment, done here: #20958